tests/org.eclipse.swt.tests.win32/JUnit Tests/org/eclipse/swt/graphics/GCWin32Tests.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Yatta Solutions
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Yatta Solutions - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import static org.junit.Assert.assertEquals;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.internal.DPIUtil;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Shell;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GCWin32Tests {
+	private Display display;
+	private int initialZoom;
+
+	@Before
+	public void setUp() {
+		initialZoom = DPIUtil.getDeviceZoom();
+		display = Display.getDefault();
+		DPIUtil.setDeviceZoom(100);
+	}
+
+	@After
+	public void tearDown() {
+		DPIUtil.setDeviceZoom(initialZoom);
+	}
+
+	@Test
+	public void gcZoomLevelMustChangeOnShellZoomChange() {
+		int zoom = DPIUtil.getDeviceZoom();
+		Shell shell = new Shell(display);
+		shell.addListener(SWT.Paint, event -> {
+			assertEquals("GCData must have a zoom level equal to the actual zoom level of the widget/shell", shell.zoom, event.gc.getGCData().deviceZoom);
```

Comment:
Don't use assertions inside listeners, they will only be printed to the output. If you change this to `assertTrue(false)` then this test still passes.